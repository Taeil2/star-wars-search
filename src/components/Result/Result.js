import React from 'react';

export default class Result extends React.Component {
  render() {
    console.log(this.props);

    return (
      <div className='result'>
        <h2>{this.props.character.name}</h2>
        <p>
          Gender: {this.props.character.gender}<br/>
          Born {this.props.character.birth_year}<br/>
          Homeworld: {this.props.character.homeworldName}
        </p>
      </div>
    );
  }
}
