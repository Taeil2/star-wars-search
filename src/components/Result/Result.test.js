import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Result from './Result';
import './Result.scss';

// smoke test
it('renders without crashing', () => {
  let result = {
    name: 'Luke Skywalker',
    gender: 'male',
    birth_year: '19BBY',
    homeworldName: 'Tattoine'
  };
  const div = document.createElement('div');
  ReactDOM.render(<Result character={result} key={0} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it('renders the UI as expected', () => {
  let result = {
    name: 'Luke Skywalker',
    gender: 'male',
    birth_year: '19BBY',
    homeworldName: 'Tattoine'
  };
  const tree = renderer
    .create(<Result character={result} key={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
