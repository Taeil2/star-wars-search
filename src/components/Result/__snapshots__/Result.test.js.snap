// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="result"
>
  <h2>
    Luke Skywalker
  </h2>
  <p>
    Gender: 
    male
    <br />
    Born 
    19BBY
    <br />
    Homeworld: 
    Tattoine
  </p>
</div>
`;
